### DISCUSSION TITLE
Help with Mp3tag for Mac, Appdata Files

### DISCUSSION THREAD

**Discussion by sanpedrodeatacama** [seanap/Plex-Audiobook-Guide#71] [Q&A]
### Goal: 
Seeking anyone with a little patience and know-how for running this process on Mp3tag's Mac version (or other simple alternatives).  

### Background: 
My files are .m4b audiobooks with clean names like "Book—Author.m4b", but they also have incorrectly assigned tags that are messing up the Audnexus agent when added to Plex (this is without doing the Mp3tag process). The Plex folders are correctly structured as `Audiobooks/Author/Book/Book.m4b`. 

I have no coding or related tech experience, and have been trying to figure this out for a couple days through Github, Youtube, and subreddits to no avail. I looked at book tagging alternatives, but they proved too complicated for me to even install (Bragibooks, Beets, and Libation). I tried MusicBrainz Picard for Mac, but it doesn't seem to have many book authors in its database.

### Current Hurdle in Plex-Audiobook-Guide: 

From the readme pictures and steps, it seems that Mp3tag varies greatly between the Windows and Mac versions. For example, in the Mp3tag Actions drop-down picture under [Test](https://github.com/seanap/Plex-Audiobook-Guide#test), I don't have any of those options to "Rename Relocate Extras Title". I just have "Actions" and "Quick Actions" ([screenshot](https://imgur.com/a/exBxVyK)).

#### So far I have:

- Downloaded Mp3tag for Mac and designated a temp folder for it to pull from. 
- Installed Audnexus agent on Plex and configured it.
- Downloaded the Plex-Audiobook-Guide-master folder. 

But my hurdle is at the step of altering Mp3tag's Appdata directory. Specifically, "[Download my example configuration files to Mp3tag's Appdata directory](https://github.com/seanap/Plex-Audiobook-Guide#download-my-example-configuration-files-to-mp3tags-appdata-directory)". 

I can't find anything on my Mac under "Appdata" or "Roaming" to merge/overwrite with the mp3tag folder. I think I'm capable of editing the config files with my specific file paths, but just can't find the right folder to merge/overwrite. According to [this](https://ourtechroom.com/tech/applications-data-folder-mac/) article, AppData on Macs is located in 3 places (Application Support, Caches, and Preferences Folders). So which one should I be finding and overwriting?

Any guidance, links to resources, really any help would be hugely appreciated. As a big fan of Plex, Prologue, and audiobooks, figuring out book tags is my final hurdle to utilizing the Audnexus agent and being able to have an organized library. Thank you.

**Comment by seanap** [comment]
The mac version of mp3tag does not support the Audible web source.  This means that natively you can't pull tags from Audible.  Some users have had success with running the windows version in either wine or a vm. https://github.com/seanap/Plex-Audiobook-Guide/issues/2#issuecomment-713587730

The action scripts should still be compatible, but the guide was written from the Windows version so the appdata folder location is just a windows thing.  I don't have any macs to test, but wherever the program data for mp3tag is stored is where the action files can go. However;

The action scripts that I wrote are meant to take the audible tags and move your book into the right folder structure with folder names based on the audible tags.  Your situation is a little unique, you already have good folder names but bad tags.  You may be better off to use the built in function (button on the ribbon): `Convert>Filename - Tag` Then enter `%artist%\%album%\%title%` and mp3tag will auto tag the ARTIST and ALBUM based on foldernames.

![image](https://user-images.githubusercontent.com/17012946/160617113-b6d2d360-b8b6-45ac-88b2-6488b9d4fd83.png)

**Reply by sanpedrodeatacama** [reply]
Wow thanks so much for this. All the little pieces of advice help a lot. Knowing this version doesn't support the Audible web source is great to know. I've since found that Audnexus doesn't need too much to go off of, and have set up some Actions with your advice. Thanks again.

**Reply by ElysiumFic** [reply]
In a recent update, I believe Mp3Tag for Mac has started working with external tag sources.

If you go into Tag Sources you should now see a "Open Tag Sources Folder" which will take you to where you need to copy the ".src" files to.

Most of the Action scripts should be importable through the Mp3Tag Actions UI. It's Actions>Actions>(downward carrot between the + and - icons beneath the menu on the left. One of those options should be "Import..." and you can select your .mta files. However, the big "Rename Relocate Extras Title" file won't import. 

I assume this is because that script contains a command that isn't supported by Mp3Tag for Mac. Furthermore, I suspect the commands that aren't working are the ones that start with "T=15" when you look at the .mta file in a text editor. Those are the ones that export the desc.txt and reader.txt files. If you can live without those files, you could conceivably remove those commands. Or create a new script in the UI that only implements the other commands.

I don't believe Mp3Tag for Mac has Export functionality yet. And the place where the files are stored is really deep and obscure in the file system. At least, the `Sources` directory is. I haven't found the equivalent of the `Export` and `Data>Actions` directories yet.
